### PR TITLE
Conf patch force exit when conf file passed on command line and not found

### PR DESCRIPTION
**Pull request by jdodgen**
Removed extra %s causing segfault and improved my changes

### CODE REVIEW

**Comment by Mr-Dave** [comment]
It is hard to spot the changes with the tabs/spaces.  Is this going to no longer search other directories if the file isn't found?

**Comment by tosiara** [comment]
Agree, whitespace changes makes it harder to understand the change.
@jdodgen if you have spare time you could add more commits to your branch to revert whitespace changes

File: conf.c [hunk 1]
```diff
@@ -1582,26 +1582,26 @@ static void conf_cmdline(struct context *cnt, int thread)
                 cnt->log_level = (unsigned int)atoi(optarg);
             break;
         case 'k':
-	  if (thread == -1) {
```
Comment (tosiara) [inline conf.c]: Whitespace change

File: conf.c [hunk 2]
```diff
@@ -1582,26 +1582,26 @@ static void conf_cmdline(struct context *cnt, int thread)
...
-	    cnt->log_type_str[sizeof(cnt->log_type_str) - 1] = '\0';
-	  }
-	  break;
+            if (thread == -1) {
+                strncpy(cnt->log_type_str, optarg, sizeof(cnt->log_type_str) - 1);
+                cnt->log_type_str[sizeof(cnt->log_type_str) - 1] = '\0';
+            }
+            break;
         case 'p':
-	  if (thread == -1) {
```
Comment (tosiara) [inline conf.c]: Whitespace change

File: conf.c [hunk 3]
```diff
@@ -1582,26 +1582,26 @@ static void conf_cmdline(struct context *cnt, int thread)
                 cnt->log_level = (unsigned int)atoi(optarg);
             break;
         case 'k':
```
Comment (tosiara) [inline conf.c]: Whitespace changes start here

File: conf.c [hunk 4]
```diff
@@ -1582,26 +1582,26 @@ static void conf_cmdline(struct context *cnt, int thread)
...
-	  break;
+            if (thread == -1) {
+                strncpy(cnt->log_file, optarg, sizeof(cnt->log_file) - 1);
+                cnt->log_file[sizeof(cnt->log_file) - 1] = '\0';
+            }
+            break;
         case 'm':
             cnt->pause = 1;
-            break;    
+            break;
```
Comment (tosiara) [inline conf.c]: and end here
Would be great to revert them back

File: conf.c [hunk 5]
```diff
@@ -1928,7 +1932,7 @@ struct context **conf_load(struct context **cnt)
       strncpy(cnt[0]->conf_filename, filename, sizeof(cnt[0]->conf_filename) - 1);
       cnt[0]->conf_filename[sizeof(cnt[0]->conf_filename) - 1] = '\0';
       MOTION_LOG(NTC, TYPE_ALL, NO_ERRNO, "%s: Processing thread 0 - config file %s",
```
Comment (tosiara) [inline conf.c]: also unnecessary change

**Comment by jdodgen** [comment]
I am away from my development systems, using a very unreliable DSL line.  I'l do this in about a week.  I do need a clarification, do you want comments around the fixed white space stuff or just to undo the white space changes?  I just thought that because it was important for me to fix my tabs it would be a good thing to correct the others.

**Comment by tosiara** [comment]
You should fix back space/tab changes so it remains unchanged if compared with previous file version. That means, if you "diff" your changes either in console or in Github - you should not see those lines as changed (red/green)

**Comment by jdodgen** [comment]
Ok, no problem

**Comment by jdodgen** [comment]
whitespace fixes have been reversed.

**Comment by Mr-Dave** [comment]
This changes the functionality and I think that if it is implemented, there would be implications to other users.  I seem recall seeing in the motion user list or other discussion boards where users were counting upon this falling through to search if the command line option was not found.  If the community believes that it should abort instead of search, I'll revive the pull.
